frontend/src/views/instructor/routes/index.tsx

```diff
@@ -1,21 +1,46 @@
 import React from "react";
-import { Redirect, Route, Switch } from "react-router-dom";
+import { Redirect, Route, Switch, useParams } from "react-router-dom";
 import { InstructorAssignmentsView } from "../assignments";
 import { Landing as InstructorLanding } from "../landing";
+import { InstructorDdahsView } from "../ddahs";
+import { useThunkDispatch } from "../../../libs/thunk-dispatch";
+import { setActivePositionId } from "../store/actions";
+
+/**
+ * React component that will update the active position id in the
+ * Redux store whenever it is rendered. This component is meant to be the first
+ * component in a route that depends on an active position.
+ */
+function UpdateActivePosition() {
+    const params = useParams<{ position_id?: string }>();
+    const dispatch = useThunkDispatch();
+    const rawPositionId = +(params.position_id || NaN);
+    const position_id = Number.isFinite(rawPositionId) ? rawPositionId : null;
+    React.useEffect(() => {
+        dispatch(setActivePositionId(position_id));
+    }, [position_id, dispatch]);
+    return null;
+}
 
 export function InstructorRoutes() {
     return (
         <Switch>
             <Route exact path="/">
                 <InstructorLanding />
             </Route>
-            <Route path="/:positionId/assignments">
+            <Route path="/positions/:position_id/assignments">
```

Comment:
Will this route work with the header we currently have?

Reply:
With the fixed header code, it works!